packages/core/conductor-tasks/common/lockS3File.ts

```diff
@@ -0,0 +1,83 @@
+import type { ConductorWorker } from "@io-orkes/conductor-javascript"
+import completed from "@moj-bichard7/common/conductor/helpers/completed"
+import failed from "@moj-bichard7/common/conductor/helpers/failed"
+import failedTerminal from "@moj-bichard7/common/conductor/helpers/failedTerminal"
+import inputDataValidator from "@moj-bichard7/common/conductor/middleware/inputDataValidator"
+import type Task from "@moj-bichard7/common/conductor/types/Task"
+import createS3Config from "@moj-bichard7/common/s3/createS3Config"
+import readTagsFromS3 from "@moj-bichard7/common/s3/readTagsFromS3"
+import writeTagsToS3 from "@moj-bichard7/common/s3/writeTagsToS3"
+import { isError } from "@moj-bichard7/common/types/Result"
+import { z } from "zod"
+
+const s3Config = createS3Config()
+
+const inputDataSchema = z.object({
+  fileName: z.string(),
+  bucketId: z.string(),
+  workflowId: z.string()
+})
+type InputData = z.infer<typeof inputDataSchema>
+
+const { TASK_DATA_BUCKET_NAME } = process.env
+if (!TASK_DATA_BUCKET_NAME) {
+  throw Error("TASK_DATA_BUCKET_NAME environment variable is required")
+}
+
+const buckets: Record<string, string> = {
+  "task-data": TASK_DATA_BUCKET_NAME
+}
+
+const lockKey = "lockedByWorkstream"
+
+const deleteS3File: ConductorWorker = {
```

Comment:
Copy and pasta - I think this should be `lockS3File`? 😝

Reply:
Good spot, thanks!